test/Constraints/diagnostics.swift

```diff
@@ -764,4 +764,13 @@ func r21523291(_ bytes : UnsafeMutablePointer<UInt8>) {
   let r = bytes[i++]  // expected-error {{cannot pass immutable value as inout argument: 'i' is a 'let' constant}}
 }
 
+// SR-1594: Wrong error description when using === on non-class types
+class SR1594 {
+  func sr1594(bytes : UnsafeMutablePointer<Int>, _ i : Int?) {
+    _ = (i === nil) // expected-error {{value of type 'Int?' does not have reference semantics, comparison isn't allowed}}
```

Comment:
This was the original bug case. It would be nice if it generated a fixit too, since it seems like the most common way that someone could go wrong.